spring-cloud-commons/src/test/java/org/springframework/cloud/commons/publisher/FluxFirstNonEmptyEmittingTests.java

```diff
@@ -172,4 +181,191 @@ public void scanRaceCoordinator() {
 		assertThat(parent.scan(Scannable.Attr.CANCELLED)).isTrue();
 	}
 
+	static class LambdaSubscriber<T> implements CoreSubscriber<T>, Scannable, Disposable {
```

Comment:
`LambdaSubscriber` is used in core tests because A) we need to pass a non-null `Subscriber` to the relevant constructors and B) we need something that is implementing `Scannable`.

Rather than copy the entire class, one solution would be to have a child class that extends `BaseSubscriber<T>` and make it implement `Scannable`. The `scanUnsafe` method needs to be overridden, but can simply `return null;`. This would greatly simplify this part, from 200 LoC to about 10.